auth/src/main/java/org/aerogear/mobile/security/metrics/SecurityCheckResultMetric.java

```diff
@@ -48,19 +66,32 @@ public String identifier() {
      * the value is <code>true</code> if the check result passed
      */
     @Override
-    public Map<String, String> data() {
-        return Collections.unmodifiableMap(data);
+    public JSONArray data() {
+        // TODO: consider returning a deep clone
```

Comment:
I would not create the JSONArray during instantiation but rather store the results as class field and return a new JSONArray each time:
```java
    public SecurityCheckResultMetric(@NonNull final Iterable<SecurityCheckResult> results) {
        this.results = results;
    }

    public SecurityCheckResultMetric(@NonNull final SecurityCheckResult... results) {
        this.results = Arrays.asList(results);
    }

    public JSONArray data() {
        return getDataFromResult(this.results);
    }
```
Also I'd add a test to ensure data cannot be changed if this would be a problem.